actionpack/test/dispatch/request_test.rb

```diff
@@ -42,9 +42,6 @@ class RequestUrlFor < BaseRequestTest
 
     assert_equal "/books", url_for(only_path: true, path: "/books")
 
-    assert_equal "http://www.example.com/books/?q=code", url_for(trailing_slash: true, path: "/books?q=code")
```

Comment:
Why were those tests removed?

Reply:
Because I believe they test something that's the underlying implementation an not public API https://github.com/rails/rails/pull/8704

Assuming that `url_for(path: "/foo")` isn't public API (it isn't documented as far as I can tell).

Reply:
But why not delete the test above that? I was not sure if it was because the behavior changed and the test failed, or just because we were cleaning up implementation detail tests.

In general we test non public API in the rails test suite if it is necessary, but if it is being tested using the public API it is fine to remove.

Reply:
> But why not delete the test above that as well? I was not sure if it was because the behavior changed and the test failed

Right. These two failed because they were testing the feature through a private API.

Reply:
So there is a change in behavior? I find that concerning. Would this change reintroduce the behavior #8704 was supposed to fix?

Reply:
No, I don't think so. But I feel like we're talking past each others, it's my fault I didn't take the time to properly braindump in the PR description, I'll do that.

Reply:
@rafaelfranca I added some more context to the PR and also added some more integration style tests.